dvc/command/add.py

```diff
@@ -74,6 +77,24 @@ def add_parser(subparsers, parent_parser):
         help="Specify name of the DVC-file this command will generate.",
         metavar="<filename>",
     )
+    parser.add_argument(
+        "-o",
+        "--out",
+        help="Destination path to put files to.",
+        metavar="<filename>",
+    )
+    parser.add_argument(
+        "--to-remote",
+        action="store_true",
+        default=False,
+        help="Download it directly to the remote",
+    )
+    parser.add_argument(
+        "-r",
+        "--remote",
+        help="Remote storage to download to",
```

Comment:
I think this should definitely be included in `--to-remote` (let it accept a remote name as argument) to simplify the UI. `add --to-remote --remote` looks fort of repetitive anyway.

Reply:
We actually though about it and decided for the sake of consistency with other commands who works on the remote, it would better to just split it out.

Reply:
also, in most cases it'll be just `--to-remote`, right? 

also, can an option in argparse be made to optionally accept a value - in this case we would have to make `--to-remote [remote-name]` where remote-name is optional (default remote if not specified)

Reply:
> also, in most cases it'll be just --to-remote, right?

yes

> also, can an option in argparse be made to optionally accept a value - in this case we would have to make --to-remote [remote-name] where remote-name is optional (default remote if not specified)

yes, the implementation is possible.

Reply:
> yes, the implementation is possible.

then, I guess, I don't have a strong preference I think. `--remote` is good for consistency, but I would not say it looks critical to me, `--to-remote` already reminds it a lot.

Reply:
> also, in most cases it'll be just --to-remote, right?

I'm not sure that most users employ default remotes. Do we know this?

I see the consistency argument but it's not that big a deal, and forcing the exact flag name causes other issues e.g. just `add --remote` is meaningless (and forbidden, I think, or it it silently ignored?) + an extra flag to document and explain 😋

Reply:
> I'm not sure that most users employ default remotes. Do we know this?

that's why we have default remotes at all. if we are not sure about this we should reconsider the whole logic we work with remotes then

> add --remote is meaningless 

agreed, I would not do this. may be keep only `--to-remote <optional-name>`